### PR TITLE
Fix EFR32 unit test

### LINKED ISSUE #10447: EFR32 Unit Test crashes
#### Problem
EFR32 Unit tests crashes while running:
`python -m nl_test_runner.nl_test_runner -d /dev/ttyACM0 -f out/debug/chip-efr32-device_tests.s37 -o out.log`

```
...
<detail> [IN] New secure session created for device 0x0000000006A11E3D, key 2!!
<detail> [IN] New secure session created for device 0x000000000001E306, key 1!!
<info  > [IN] Prepared encrypted message 0x20020518 to 0x0000000006A11E3D of type 0xhx and protocolId (1, 0) on exchange 4� with
 MessageCounter:105.
<info  > [IN] Sending encrypted msg 0x20020518 with MessageCounter:1 to 0x0000000006A11E3D at monotonic time: ld msec
<info  > [-] 
FREERTOS ASSERT ( (pxLink->xBlockSize & xBlockAllocatedBit) != 0 )

<info  > [-] 
FREERTOS ASSERT ( ( portNVIC_INT_CTRL_REG & portVECTACTIVE_MASK ) == 0 )
```

### PR DESCRIPTION
**Pull request by rgoliver**
#### Problem
The EFR32 Unit test has been broken for almost 2 weeks: #10447 

#### Change overview
- Remove broken tests so it passes
- Add to build script so it gets compiled into CI
- Fix nlrunner to stream output
- Bundle in runner whls into output
- Add to smoke-test (Also fixed smoke test globs to pull in rpc builds for EFR and NRF)

#### Testing
- Ran EFR32 unit test
- Ran script/build/test.py
- Ran smoke test command and ensured all targets got built correctly:
`/scripts/build/build_examples.py --target-glob '*-brd4161a-{lock,light,unit-test}*' build `

### CODE REVIEW

**Review by bzbarsky-apple — DISMISSED** [review]

File: src/BUILD.gn [hunk 1]
```diff
@@ -73,6 +72,7 @@ if (chip_build_tests) {
 
     if (chip_device_platform != "efr32") {
       deps += [
+        "${chip_root}/src/app/tests",
```
[the remark this answers is not in the document]
Comment (rgoliver) [reply src/BUILD.gn:78]: Done

File: src/BUILD.gn [hunk 2]
```diff
@@ -96,7 +96,8 @@ if (chip_build_tests) {
 
     # On nrfconnect, the controller tests run into
     # https://github.com/project-chip/connectedhomeip/issues/9630
-    if (chip_device_platform != "nrfconnect") {
+    if (chip_device_platform != "nrfconnect" &&
+        chip_device_platform != "efr32") {
```
[the remark this answers is not in the document]
Comment (rgoliver) [reply src/BUILD.gn:103]: Done

File: src/messaging/tests/BUILD.gn [hunk 3]
```diff
@@ -45,9 +45,12 @@ chip_test_suite("tests") {
   sources = [
     "TestExchangeMgr.cpp",
     "TestMessagingLayer.h",
-    "TestReliableMessageProtocol.cpp",
   ]
 
+  if (chip_device_platform != "efr32") {
+    sources += [ "TestReliableMessageProtocol.cpp" ]
```
[the remark this answers is not in the document]
Comment (rgoliver) [reply src/messaging/tests/BUILD.gn:53]: Done

File: src/app/tests/BUILD.gn [hunk 4]
```diff
@@ -44,24 +44,19 @@ chip_test_suite("tests") {
     "TestAttributeValueEncoder.cpp",
```
[the remark this answers is not in the document]
Comment (rgoliver) [reply src/app/tests/BUILD.gn:44]: The app tests have dependencies on eachother, which are causing linking errors and frequent breakages when only some are removed and not others. For now just removing them all and prioritizing this as one of the first to get working again.

File: src/BUILD.gn (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmeg-sfy) [reply src/BUILD.gn:103]: You mention https://github.com/project-chip/connectedhomeip/issues/9630 but it seems to be resolved now !?

[the remark this answers is not in the document]
Comment (jmeg-sfy) [reply src/BUILD.gn:103]: also then you should add TODOs to be able to track down those and re-enable them later

[the remark this answers is not in the document]
Comment (rgoliver) [reply src/BUILD.gn:102]: #9630 is the comment for nrf, #10447 is for EFR32 (comment a couple lines down). updated them all to TODOs

**Review by Damian-Nordic — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**